File: sdks/csharp/flapjacksearch/Exceptions/FlapjackApiException.cs
using System;

namespace Flapjack.Search.Exceptions;

/// <summary>
/// Exception sent by Flapjack's API
/// </summary>
public class FlapjackApiException : Exception
{
  /// <summary>
  /// Http error code
  /// </summary>
  public int HttpErrorCode { get; set; }

  /// <summary>
  /// Create a new FlapjackAPIException
  /// </summary>
  /// <param name="message"></param>
  /// <param name="httpErrorCode"></param>
  public FlapjackApiException(string message, int httpErrorCode)
    : base(message)
  {
    HttpErrorCode = httpErrorCode;
  }
}

File: sdks/java/flapjacksearch/src/main/java/com/flapjackhq/exceptions/FlapjackApiException.java
package com.flapjackhq.exceptions;

/** Exception thrown in case of API failure such as 4XX, 5XX error. */
public class FlapjackApiException extends FlapjackRuntimeException {

  private static final long serialVersionUID = 1L;

  public int getStatusCode() {
    return statusCode;
  }

  private final int statusCode;

  public FlapjackApiException(String message, Throwable cause, int httpErrorCode) {
    super(message, cause);
    this.statusCode = httpErrorCode;
  }

  public FlapjackApiException(String message, int httpErrorCode) {
    super(message);
    this.statusCode = httpErrorCode;
  }

  public FlapjackApiException(Throwable cause, int httpErrorCode) {
    super(cause);
    this.statusCode = httpErrorCode;
  }

  @Override
  public String getMessage() {
    String message = super.getMessage();
    return "Status Code: " + getStatusCode() + (message != null ? " - " + message : "");
  }
}
